sweepbatcher/sweep_batch.go

```diff
@@ -1908,11 +1910,40 @@ func (b *batch) handleSpend(ctx context.Context, spendTx *wire.MsgTx) error {
 
 // handleConf handles a confirmation notification. This is the final step of the
 // batch. Here we signal to the batcher that this batch was completed.
-func (b *batch) handleConf(ctx context.Context) error {
+func (b *batch) handleConf(ctx context.Context,
+	conf *chainntnfs.TxConfirmation) error {
+
```

Comment:
Since this function iterates on sweeps, we should also schedule it to the main eventloop.

Reply:
`handleConf` is already called from the main event loop, so it can freely access `batch.sweeps` - nobody else should access it at the same time.

Reply:
Indeed, you're right.